stock_location_bin_name/models/stock_location.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2017 Syvain Van Hoof (Okia sprl) <sylvainvh@okia.be>
+# Copyright 2016-2019 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import models, fields, api
+
+
+class StockLocation(models.Model):
+
+    _inherit = 'stock.location'
+
+    location_name_format = fields.Char(
+        'Location Name Format',
+        help="Format string that will compute the name of the location. "
+             "Use location fields. Example: "
+             "'{area}-{corridor:0>2}.{rack:0>3}"
+             ".{level:0>2}'")
+
+    area = fields.Char(
+        'Area',
+        compute='_compute_area',
+    )
+
+    @api.depends('name', 'location_kind', 'location_id.area')
+    def _compute_area(self):
+        for location in self:
+            if location.location_kind == 'area':
+                location.area = location.name
+            else:
+                location.area = location.location_id.area
```

Comment:
IMHO we can eliminate the else part by defining line 30 before if condition.

Reply:
It would access `location.location_id` (so read another record) even when not required.